pkg/controllers/user/pipeline/controller/project/project.go

```diff
@@ -0,0 +1,99 @@
+package project
+
+import (
+	"context"
+	"github.com/rancher/rancher/pkg/pipeline/remote/model"
+	"github.com/rancher/rancher/pkg/pipeline/utils"
+	"github.com/rancher/rancher/pkg/ref"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	pv3 "github.com/rancher/types/apis/project.cattle.io/v3"
+	pclient "github.com/rancher/types/client/project/v3"
+	"github.com/rancher/types/config"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var settings = map[string]string{
+	utils.SettingExecutorQuota: utils.SettingExecutorQuotaDefault,
+}
+
+func Register(ctx context.Context, cluster *config.UserContext) {
+	projects := cluster.Management.Management.Projects("")
+	projectSyncer := &Syncer{
+		sourceCodeProviderConfigs: cluster.Management.Project.SourceCodeProviderConfigs(""),
+		pipelineSettings:          cluster.Management.Project.PipelineSettings(""),
+	}
+
+	projects.AddClusterScopedHandler("pipeline-controller", cluster.ClusterName, projectSyncer.Sync)
+}
+
+//Syncer is responsible for watching projects and
+// maintaining initialized source code provider configs & pipeline settings
+type Syncer struct {
+	sourceCodeProviderConfigs pv3.SourceCodeProviderConfigInterface
+	pipelineSettings          pv3.PipelineSettingInterface
+	clusterName               string
+}
+
+func (l *Syncer) Sync(key string, obj *v3.Project) error {
+	if obj == nil {
```

Comment:
should we check for not nil object having DeletionTimestamp != nil? If yes, could you revise the rest of the code for similar check